exporter/collector/config.go

```diff
@@ -48,6 +48,17 @@ type ClientConfig struct {
 
 type TraceConfig struct {
 	ClientConfig ClientConfig `mapstructure:",squash"`
+	// AttributeKeyMappings determines how to map from OpenTelemetry attribute keys to
+	// Google Cloud Trace keys.  By default, it changes keys so that they appear more
+	// prominently in the UI.
+	AttributeKeyMappings []AttributeKeyMapping `mapstructure:"attribute_mappings"`
```

Comment:
Why do we use a list of pairs representation, when a map would be more concise and document as well as ensure that keys are distinct?

Reply:
We could switch to maps.  The current way this would look in config is:

```yaml
attribute_mappings:
- key: service.name
  replacement: different.service.key
- key: foo
  replacement: bar
```

If we used map[string]string it would look like:

```yaml
attribute_mappings:
  service.name: different.service.key
  foo: bar
```

But that doesn't allow the API to evolve, since you can't add additional fields to the value.

We could allow evolution with a sub-object if we used map[string]struct{replacement: string}, and it would look like:

```yaml
attribute_mappings:
  service.name:
    replacement: different.service.key
  foo:
    replacement: bar
```

Of the three, I found the first to be the easiest to read (but I can see your case for the second).  You are right that I should be validating the uniqueness of the keys.  Right now the behavior is that the last key takes precedence, but returning an error to the user would be better.

Reply:
Supporting evolution is a great reason to rule out `map[string]string`, thanks for calling that out. I'm agnostic between list-of-pairs and map-with-struct-values; though I'd request some validation of keys if we go with the list representation.

Reply:
added validation